source/extensions/access_loggers/wasm/config.cc

```diff
@@ -20,30 +20,29 @@ namespace AccessLoggers {
 namespace Wasm {
 
 AccessLog::InstanceSharedPtr
-WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
+WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& proto_config,
                                               AccessLog::FilterPtr&& filter,
                                               Server::Configuration::FactoryContext& context) {
-  const auto& wal_config =
-      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(config);
+  const auto& config =
+      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(
+          proto_config);
+  auto id = config.id();
+  auto configuration = std::make_shared<std::string>(config.configuration());
```

Comment:
It is being passed into a lambda which has lifetime beyond the end of this function.   The lambda is copyable and we can't take it by reference, hence the shared_ptr.

Reply:
We have to use a shared_ptr for 2 reasons: 1) envoy is currently on c++14 which doesn't support move-only std:;functions with unique_ptr<>s.  2) this lambda must be copyable as it is passed to each thread by copy (and it isn't a downward funarg because the lambda outlives this function invocation).